test/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,85 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-unnecessary-polyfills';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+ruleTester.run('no-unnecessary-polyfills', rule, {
+	valid: [
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '0.1.0'}]
+		},
+		{
+			code: 'import assign from "object-assign"',
+			options: [{targetVersion: '0.1.0'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '<4'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '>3'}]
+		}
+	],
+	invalid: [
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'import assign from "object-assign"',
+			options: [{targetVersion: '6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '>6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '8'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("array-from")',
+			options: [{targetVersion: '>7 <8'}],
```

Comment:
This does not actually match any ranges. `>7` is treated as `>=8.0.0` by semver and the range `>=8.0.0 <8` cannot possibly match any version.

I think this should be moved to the `valid` list, `require("array-from")` should be allowed for this range.

I've opened a bug at npm/node-semver#275 to document this edge case of semver ranges.